### PR TITLE
Ammo check animations

### PR DESCRIPTION
**Pull request by Utjan**
Looks for animations: anm_ammo_check, anm_ammo_check_empty, anm_ammo_check_no_mag
Sounds: snd_ammo_check, snd_ammo_check_empty, snd_ammo_check_no_mag

### CODE REVIEW

**Comment by Ishmaeel** [comment]
Where are those animations and sounds found?

**Comment by Utjan** [comment]
> Where are those animations and sounds found?

In custom made weapons that have them. I made this script edit for Barry's VSSK mod. If a weapon doesn't have the animations, it does the normal ammo check.

**Comment by Ishmaeel** [comment]
Ooh, I see now, very nice.

However, you will agree that this is a significant change with very few applications. I cannot justify merging this to the main branch until many weapon packs support those animations. Yes, the added code degrades gracefully and does not seem to break anything, but it still comes with a maintenance cost.

That said, I might add some callbacks/extensibility points where other addons can attach to. This way, custom animation code would live in its own script file and everything would integrate without being too tightly coupled.

I will consider this.

**Comment by Utjan** [comment]
Ah okay. That's fair I guess.
Callbacks or something patchable would be nice then.

File: gamedata/scripts/ammo_check_mcm.script [hunk 1]
```diff
@@ -279,24 +287,76 @@ function check_Ammo()
...
-                weapon_hidden = false
-                return true
-            end
-            actor_menu.set_msg(1, message, 2, clr)
-            db.actor:activate_slot(slot)
-
-            return true
-        end
-            , slot)
+		local anm_ammo_check = SYS_GetParam(0, weapon:section() .. "_hud", "anm_ammo_check")
```
Comment (ahuyn) [inline gamedata/scripts/ammo_check_mcm.script:290]: Recommend to pull from `hud` parameter

**Pull request merged**